src/App.tsx

```diff
@@ -1,92 +1,68 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useContext, useState } from 'react';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
+import { TodoContext } from './context/TodoContext';
 
 export const App: React.FC = () => {
+  const { todos, setTodos, addTodo } = useContext(TodoContext);
+
+  const [query, setQuery] = useState('');
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    addTodo(query);
+    setQuery('');
+  };
+
+  const handleMakeAllCompletedTodo = () => {
+    const areMostTodosCompleted
```

Comment:
It could be implemented with some or every method.  in my opinion it would be more understandable.